src/zmq_helpers.cpp

```diff
@@ -42,11 +42,14 @@ message_t::message_t(void* data, size_t size, void (*free_function)(void*, void*
     delete message;
   });
 }
-message_t::message_t(size_t size) {
+message_t::message_t(size_t size, const void* data) {
   // make the c message
   zmq_msg_t* message = new zmq_msg_t();
   if (zmq_msg_init_size(message, size) != 0)
     throw std::runtime_error(zmq_strerror(zmq_errno()));
+  // copy the data into it
+  if (size && data)
+    memcpy(zmq_msg_data(message), data, size);
```

Comment:
yeah so before this constructor didnt do anything it was like the default make a message that is empty or random junk initialized but of a certain size. now you can actually pass data to it to manage